src/pydata_sphinx_theme/toctree.py

```diff
@@ -101,8 +101,18 @@ def unique_html_id(base_id: str):
         return next(get_or_create_id_generator(base_id))
 
     @cache
-    def generate_header_nav_before_dropdown(n_links_before_dropdown):
-        """The cacheable part."""
+    def generate_header_nav_before_dropdown(
+        n_links_before_dropdown,
+    ) -> Tuple[str, List[str]]:
+        """Return html for navbar and dropdown.
+
+        Given the number of links before the dropdown, return the html for the navbar,
+        as well as the list of links to but in a dropdown.
+
+        Returns:
+            - HTML str for the navbar
+            - list of HTML str for the dropdown
+        """
```

Comment:
Is there a way we could better communicate to other devs that this function is not really meant to be used on its own? It's just a memoized sub-procedure to help speed up build time.

Reply:
If it were me I'd prefix the func name with `_` and add a short docstring saying "don't call this directly" or so